ibridgesgui/search.py

```diff
@@ -0,0 +1,176 @@
+        self.searchResTable.show()
+        self.downloadButton.show()
+        self.clearButton.show()
+        self.info_label.show()
+
+    def clear(self):
+        """Crea search results and hide elements."""
+        self.errorLabel.clear()
+        self.hide_result_elements()
+
+    def search(self):
+        """Validate search parameters and search."""
+        self.hide_result_elements()
+        self.setCursor(QtGui.QCursor(QtCore.Qt.CursorShape.WaitCursor))
+        self.errorLabel.clear()
+
+        # All metadata values need a specific key
+        if any([key.text() == "" and val.text() != "" for key, val in zip(self.keys, self.vals)]):
+            self.errorLabel.setText("There are metadata values without keys. Stop search.")
+            self.setCursor(QtGui.QCursor(QtCore.Qt.CursorShape.ArrowCursor))
+            return
+
+        if all([key.text() == "" for key in self.keys]):
+            key_vals = None
+        else:
+            # Replace empty values with the wild card, turn into search key_vals
+            key_vals = dict(zip([key.text() for key in self.keys],
+                                [ "%" if val.text() == "" else val.text() for  val in self.vals]))
+            del key_vals['']
+        path = self.path_field.text() if self.path_field.text() != "" else None
+        checksum = self.checksum_field.text() if self.checksum_field.text() != "" else None
+
+        if key_vals is None and path is None and checksum is None:
+            self.errorLabel.setText("No search critera given.")
+        else:
+            results = search_data(self.session, path=path, checksum=checksum, key_vals=key_vals)
+            if len(results) == 0:
+                self.errorLabel.setText("No objects or collections found.")
+            else:
+                self.show_result_elements()
+                self.load_results(results)
+        self.setCursor(QtGui.QCursor(QtCore.Qt.CursorShape.ArrowCursor))
+
+    def load_results(self, results):
+        """Load seach results into the table view."""
+        table_data = [] # (Path, Name, Size, Checksum, created, modified)
+        for result in results:
+            if "DATA_NAME" in result:
+                obj = get_dataobject(self.session, result["COLL_NAME"]+"/"+result["DATA_NAME"])
+                table_data.append(('-d', obj.path, obj.size,
+                                  obj.create_time.strftime("%d-%m-%Y"),
+                                  obj.modify_time.strftime("%d-%m-%Y")))
+            else:
+                coll = get_collection(self.session, result["COLL_NAME"])
+                table_data.append(('-C', coll.path, "",
+                                  coll.create_time.strftime("%d-%m-%Y"),
+                                  coll.modify_time.strftime("%d-%m-%Y")))
+            populate_table(self.searchResTable, len(table_data), table_data)
+
+    def download(self):
+        """Download selected data."""
+        info = ""
```

Comment:
This might need a comment. Does this override the info that you imported?

Reply:
It is actually not needed ... I removed it